diff --git a/pinsketch/pinsketch/Makefile b/pinsketch/pinsketch/Makefile
index 56ff90d..69c68c2 100644
--- a/pinsketch/pinsketch/Makefile
+++ b/pinsketch/pinsketch/Makefile
@@ -11,7 +11,7 @@
 # NTL creates these when you install it according to NTL instructions.
 # See http://www.shoup.net/ntl for more.
 
-CXXFLAGS = -Wno-deprecated
+CXXFLAGS += -g -O2 -std=c++11 -pthread -march=native -Wno-deprecated -I${BSPTOOLS}/include -L${BSPTOOLS}/lib
 
 all: sketch differ
 
@@ -28,10 +28,10 @@ differ.o: differ.cpp pinsketch.h
 	g++ $(CXXFLAGS) -c differ.cpp 
 
 sketch: sketch.o bch.o io.o pinsketch.h
-	g++  $(CXXFLAGS) sketch.o io.o bch.o -lntl -o sketch
+	g++  $(CXXFLAGS) sketch.o io.o bch.o -lntl -lgmp -o sketch
 
 differ: differ.o bch.o io.o pinsketch.h
-	g++ $(CXXFLAGS) differ.o io.o bch.o -lntl -o differ
+	g++ $(CXXFLAGS) differ.o io.o bch.o -lntl -lgmp -o differ
 
 clean:
 	rm differ sketch bch.o io.o sketch.o differ.o
